src/util/HeaderUtil.ts

```diff
@@ -547,3 +552,39 @@ export function parseLinkHeader(link: string | string[] = []): LinkEntry[] {
   }
   return links;
 }
+
+/**
+ * Checks if the value of an HTTP Authorization header matches a specific scheme (e.g. Basic, Bearer, etc).
+ *
+ * @param authorization - The value of the Authorization header (may be undefined).
+ * @param scheme - Name of the authorization scheme (case insensitive).
+ * @returns True if the Authorization header uses the specified scheme, false otherwise.
+ */
+export function matchesAuthorizationScheme(authorization: string | undefined, scheme: string): boolean {
+  const lowerCaseScheme = scheme.toLowerCase();
+  if (!authSchemeRegexCache.has(lowerCaseScheme)) {
+    authSchemeRegexCache.set(lowerCaseScheme, new RegExp(`^${lowerCaseScheme} `, 'ui'));
+  }
+  // Support authorization being undefined (for the sake of usability).
+  return authorization !== undefined && authSchemeRegexCache.get(lowerCaseScheme)!!.test(authorization);
+}
+
+/**
+ * Checks if the scheme part of the specified url matches at least one of the provided options.
+ *
+ * @param url - A string representing the URL.
+ * @param scheme - The scheme that should be used to test the URL.
+ * @param schemes - Additional scheme value options (the function will check if at least one matches the URL scheme).
+ * @returns True if the URL scheme matches at least one of the provided options, false otherwise.
+ */
+export function hasScheme(url: string, scheme: string, ...schemes: string[]): boolean {
+  // Generate the cache key for the scheme options: sort the items to avoid multiple entries for the same 'check'.
+  const sortedLowerCaseSchemes = [ scheme, ...schemes ]
+    .sort((s1, s2): number => s1.localeCompare(s2))
+    .map((item): string => item.toLowerCase());
```

Comment:
Would swap these 2 so casing has no effect on the sort.

Edit: but I see below that this will be removed anyway so not relevant I guess 😄